### PR TITLE
Event aggregator

### PR DESCRIPTION
**Pull request by shreymahendru**

### CODE REVIEW

File: lib/src/event_aggregator.dart [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+/// Extends this Event for newly created App event
+///  Example:
+/// ```dart
+/// class IntEvent extends Event {
+///   final int value;
+///   IntEvent(this.value);
+/// }
+/// ```
+abstract class Event {}
```
Comment (nivinjoseph) [inline lib/src/event_aggregator.dart]: Get rid of this base class and remove all dependancy on it

File: lib/src/event_aggregator.dart [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+class FloaterEventAggregator implements EventAggregator {
+  final PublishSubject<Event> _eventPublishSubject = PublishSubject();
+  final Map<Type, Stream<Event>> _streamCache = {};
+
+  /// Subscribe to an event of a specific type.
+  /// If no type is provided, the subscriber receives all the event published
+  @override
+  Stream<T> subscribe<T extends Event>() {
+    if (T == Event) return this._eventPublishSubject.stream;
```
Comment (nivinjoseph) [inline lib/src/event_aggregator.dart]: Remove this line

File: lib/src/event_aggregator.dart [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    this._streamCache[T] ??= this._eventPublishSubject.stream.whereType<T>();
+
+    return this._streamCache[T];
+  }
+
+  /// Publish an new event.
+  @override
+  void publish(Event event) {
```
Comment (nivinjoseph) [inline lib/src/event_aggregator.dart]: Type of event should be T

**Review by nivinjoseph — CHANGES_REQUESTED** [review]

File: lib/src/event_aggregator.dart [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  /// Publish an new event.
+  @override
+  void publish<T>(T event) {
+    given(event, "event").ensureHasValue();
+    this._eventPublishSubject.add(event);
+  }
+
+  @override
+  Future<void> dispose() async {
+    await this._eventPublishSubject.close();
```
Comment (nivinjoseph) [inline lib/src/event_aggregator.dart]: You are not tracking _isDisposed and throwing exceptions appropriately in other places

**Review by nivinjoseph — CHANGES_REQUESTED** [review]

File: lib/src/event_aggregator.dart (hunk 4 above)
Comment (shreymahendru) [reply lib/src/event_aggregator.dart]: true

**Pull request merged**